FestivalPlanner/films/views.py

```diff
@@ -1,21 +1,227 @@
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        pr_debug('start', with_time=True)
+        session = self.request.session
+        self.logged_in_fan = current_fan(session)
+        self.festival = current_festival(session)
+        self.fragment_name_by_row_nr = {}
+
+        # Return the cache when valid.
+        if PickRating.film_rating_cache.is_valid(session):
+            pr_debug('done, cache is valid', with_time=True)
+            return PickRating.film_rating_cache.get_film_rows(session)
+
+        # Filter the films as requested.
+        pr_debug('start filtered query', with_time=True)
+        filter_kwargs = {'festival': self.festival}
+        if not FilmsView.display_shorts:
+            filter_kwargs['duration__gt'] = self.short_threshold
+        self.selected_films = Film.films.filter(**filter_kwargs).order_by('seq_nr')
+        self.festival_feature_films = copy.deepcopy(self.selected_films)
+        for fan in self.fan_list:
+            if not FilmsView.display_rated_by_fan[fan.name]:
+                self.selected_films = self.selected_films.filter(
+                    ~Exists(FilmFanFilmRating.film_ratings.filter(film=OuterRef('pk'), film_fan=fan))
+                )
+        film_rows = [self.get_film_row(row_nr, film) for row_nr, film in enumerate(self.selected_films)]
+
+        # Add a fragment name as to be able to address a specific film.
+        pr_debug('start defining fragments', with_time=True)
+        for row_nr, film_row in enumerate(film_rows):
+            try:
+                film_row['fragment_name'] = self.fragment_name_by_row_nr[row_nr]
+            except KeyError:
+                film_row['fragment_name'] = 0
+
+        # Fill the cache.
+        PickRating.film_rating_cache.set_film_rows(session, film_rows)
+
+        pr_debug('done', with_time=True)
+        return film_rows
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        session = self.request.session
+        pr_debug(f'start, {len(PickRating.film_rating_cache.get_film_rows(session))} records in cache', with_time=True)
+        super_context = super().get_context_data(**kwargs)
+        film_count, rated_films_count, count_dicts = get_rating_statistics(session)
+        new_context = {
+            'title': self.title,
+            'fan_headers': self.get_fan_headers(),
+            'feature_count': film_count,
+            'rated_features_count': rated_films_count,
+            'highest_rating': self.highest_rating,
+            'eligible_counts': count_dicts,
+            'short_threshold': self.short_threshold.total_seconds(),
+            'display_shorts_action': self.action_by_display_shorts[FilmsView.display_shorts],
+            'unexpected_errors': FilmsView.unexpected_errors,
+            'log': get_log(session)
+        }
+        unset_log(session)
+        context = add_base_context(self.request, {**super_context, **new_context})
```

Comment:
Please consider to update our Python version.

Among others this construction will be a bit more understandable: `super_context | new_context`

Reply:
Added #340 to do so.